addon/modifiers/on-resize.js

```diff
@@ -0,0 +1,24 @@
+import Modifier from 'ember-modifier';
+import { inject as service } from '@ember/service';
+import { action } from '@ember/object';
+
+export default class OnResizeModifier extends Modifier {
+  @service resizeObserver;
+
+  get callback() {
+    return this.args.positional[0];
+  }
+
+  didInstall() {
+    this.resizeObserver.observe(this.element, this.handleResize);
+  }
+
+  willRemove() {
+    this.resizeObserver.unobserve(this.element, this.handleResize);
+  }
+
+  @action
+  handleResize(...args) {
+    this.callback(...args);
```

Comment:
The `arguments` property in javascript is implicit even if no arguments are named in the function.
```suggestion
  handleResize() {
    this.callback(...arguments);
```